ilastik/workflows/carving/opCarving.py

```diff
@@ -540,53 +506,32 @@ def saveCurrentObjectAs(self, name):
         self.AllObjectNames.meta.shape = (len(objects),)
 
         # now that 'name' is no longer part of the set of finished objects, rebuild the done overlay
-
         self._buildDone()
-        # self._clearLabels()
-        # self._mst.clearSegmentation()
-        # self.clearCurrentLabeling()
-        # self._mst.gridSegmentor.clearSeeds()
-        # self.Trigger.setDirty(slice(None))
-        # self.updatePreprocessing()
 
     def get_label_voxels(self):
         # the voxel coordinates of fg and bg labels
         if not self.opLabelArray.NonzeroBlocks.ready():
-            return (None, None)
-
-        nonzeroSlicings = self.opLabelArray.NonzeroBlocks[:].wait()[0]
-
-        coors1 = [[], [], []]
-        coors2 = [[], [], []]
-        for sl in nonzeroSlicings:
-            a = self.opLabelArray.Output[sl].wait()
-            w1 = numpy.where(a == 1)
-            w2 = numpy.where(a == 2)
-            w1 = [w1[i] + sl[i].start for i in range(1, 4)]
-            w2 = [w2[i] + sl[i].start for i in range(1, 4)]
+            return None, None
+
+        bg = [[], [], []]
+        fg = [[], [], []]
+        for slicing in self.opLabelArray.NonzeroBlocks[:].wait()[0]:
+            label = self.opLabelArray.Output[slicing].wait()
+            labels_bg = numpy.where(label == Labels.BACKGROUND)
+            labels_fg = numpy.where(label == Labels.FOREGROUND)
```

Comment:
[Numpy recommends to use `nonzero` instead of `where`](https://numpy.org/doc/stable/reference/generated/numpy.where.html#numpy.where) if 2nd and 3rd arguments are omitted.